### PR TITLE
implemented getOutputElementType 

### PR DESCRIPTION
**Pull request by Pey-crypto**
Implemented Method on c++ side.
Updated typescript definitions.
Created unit tests.
For Issue https://github.com/openvinotoolkit/openvino/issues/25406

### CODE REVIEW

**Comment by mlukasze** [comment]
hey @Pey-crypto there is a conflict to solve, will you have a time to fix it, please?

**Comment by Pey-crypto** [comment]
Will do it soon👍

**Comment by mlukasze** [comment]
awesome, thank you in advance!

File: src/bindings/js/node/src/model_wrap.cpp [hunk 1]
```diff
@@ -171,3 +172,19 @@ Napi::Value ModelWrap::get_output_shape(const Napi::CallbackInfo& info) {
...
+            auto output = _model->output(idx);
+            return cpp_to_js<ov::element::Type_t, Napi::String>(info, output.get_element_type());
+        } else {
+            OPENVINO_THROW("getOutputElementType", ov::js::get_parameters_error_msg(info, allowed_signatures));
+        }
+    } catch (const std::exception& e) {
+        reportError(info.Env(), e.what());
+        return info.Env().Undefined();
+    }
+}
```
Comment (almilosz) [inline src/bindings/js/node/src/model_wrap.cpp:190]:
```suggestion
}

```

File: src/bindings/js/node/tests/model.test.js [hunk 2]
```diff
@@ -111,4 +111,49 @@ describe('Model.getOutputSize()', () => {
...
+
+  it('should throw an error for out-of-range index', () => {
+    const outputSize = model.getOutputSize();
+    assert.throws(
+      () => { model.getOutputElementType(outputSize); },
+      /^Error:/,
+      'Should throw for out-of-range index'
+    );
+  });
 });
```
Comment (almilosz) [inline src/bindings/js/node/tests/model.test.js:159]:
```suggestion
});

```

File: src/bindings/js/node/src/model_wrap.cpp [hunk 3]
```diff
@@ -1,12 +1,12 @@
 // Copyright (C) 2018-2024 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 
-#include "node/include/model_wrap.hpp"
-
 #include "node/include/addon.hpp"
 #include "node/include/errors.hpp"
 #include "node/include/helper.hpp"
+#include "node/include/model_wrap.hpp"
```
Comment (almilosz) [inline src/bindings/js/node/src/model_wrap.cpp:7]: Please revert this change and move `#include "node/include/model_wrap.hpp"` to the top.
Probably you are using a different formatting tool than is specified [here](https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/coding_style.md)

File: src/bindings/js/node/src/model_wrap.cpp [hunk 4]
```diff
@@ -171,3 +172,19 @@ Napi::Value ModelWrap::get_output_shape(const Napi::CallbackInfo& info) {
         return info.Env().Undefined();
     }
 }
+
+Napi::Value ModelWrap::get_output_element_type(const Napi::CallbackInfo& info) {
+    std::vector<std::string> allowed_signatures = {"(number)"};
```
Comment (almilosz) [inline src/bindings/js/node/src/model_wrap.cpp:177]:
```suggestion
    std::vector<std::string> allowed_signatures;
```
You don't have to initialize allowed signatures

File: src/bindings/js/node/src/model_wrap.cpp [hunk 5]
```diff
@@ -171,3 +172,19 @@ Napi::Value ModelWrap::get_output_shape(const Napi::CallbackInfo& info) {
...
+    try {
+        if (info.Length() == 1 && ov::js::validate_value<int>(info.Env(), info[0])) {
```
Comment (almilosz) [inline src/bindings/js/node/src/model_wrap.cpp:179]:
```suggestion
        if (ov::js::validate<int>(info, allowed_signatures)) {
```

File: src/bindings/js/node/src/model_wrap.cpp [hunk 6]
```diff
@@ -171,3 +172,19 @@ Napi::Value ModelWrap::get_output_shape(const Napi::CallbackInfo& info) {
...
+            auto idx = info[0].As<Napi::Number>().Int32Value();
+            auto output = _model->output(idx);
```
Comment (almilosz) [inline src/bindings/js/node/src/model_wrap.cpp:181]:
```suggestion
            const auto& output = _model->output(idx);
```

File: src/bindings/js/node/src/model_wrap.cpp [hunk 7]
```diff
@@ -171,3 +172,19 @@ Napi::Value ModelWrap::get_output_shape(const Napi::CallbackInfo& info) {
...
+            return cpp_to_js<ov::element::Type_t, Napi::String>(info, output.get_element_type());
+        } else {
+            OPENVINO_THROW("getOutputElementType", ov::js::get_parameters_error_msg(info, allowed_signatures));
```
Comment (almilosz) [inline src/bindings/js/node/src/model_wrap.cpp:184]:
```suggestion
            OPENVINO_THROW("'getOutputElementType'", ov::js::get_parameters_error_msg(info, allowed_signatures));
```

File: src/bindings/js/node/tests/model.test.js [hunk 8]
```diff
@@ -111,4 +111,49 @@ describe('Model.getOutputSize()', () => {
...
+  it('should return a string for the element type ', () => {
+    const result = model.getOutputElementType(0);
+    assert.strictEqual(typeof result, 'string',
+      'getOutputElementType() should return a string');
+  });
+
+  it('should accept a single integer argument', () => {
+    assert.throws(() => {
+      model.getOutputElementType();
+    }, /^Error: getOutputElementType method called with incorrect parameters\./,
```
Comment (almilosz) [inline src/bindings/js/node/tests/model.test.js:126]: You forgot to update error messages ;)

File: src/bindings/js/node/tests/model.test.js [hunk 9]
```diff
@@ -111,4 +111,49 @@ describe('Model.getOutputSize()', () => {
...
+     'Should throw when called without arguments');
+
+    assert.throws(() => {
+      model.getOutputElementType('unexpected argument');
+    }, /^Error: getOutputElementType method called with incorrect parameters\./,
```
Comment (almilosz) [inline src/bindings/js/node/tests/model.test.js:131]: .

File: src/bindings/js/node/tests/model.test.js [hunk 10]
```diff
@@ -111,4 +111,49 @@ describe('Model.getOutputSize()', () => {
...
+    'Should throw on non-number argument');
+
+    assert.throws(() => {
+      model.getOutputElementType(0, 1);
+    }, /^Error: getOutputElementType method called with incorrect parameters\./,
```
Comment (almilosz) [inline src/bindings/js/node/tests/model.test.js:136]: .

File: src/bindings/js/node/tests/model.test.js [hunk 11]
```diff
@@ -111,4 +111,49 @@ describe('Model.getOutputSize()', () => {
...
+    'Should throw on multiple arguments');
+
+    assert.throws(() => {
+      model.getOutputElementType(3.14);
+    }, /^Error: getOutputElementType method called with incorrect parameters\./,
```
Comment (almilosz) [inline src/bindings/js/node/tests/model.test.js:141]: .

**Review by almilosz — CHANGES_REQUESTED** [review]
Please apply suggestions and update tests, but it looks fine
